source/dub/generators/generator.d

```diff
@@ -439,6 +439,7 @@ struct GeneratorSettings {
 	string buildType;
 	BuildSettings buildSettings;
 	BuildMode buildMode = BuildMode.separate;
+	int targetExitStatus;
```

Comment:
is GeneratorSettings really the proper place to put this? All the other environment variables are always set after all and targetExitStatus is only set once the program has run once, so it wouldn't make sense to include it in a pre-run call for example.

Also why is this called "target" exit status?

Reply:
It seemed for me reasonable to name it "target" as e.g. the existing method also is called "runTarget". 
You are right, it is a compromise. I didn't wanted to add too much new complexity just to get the targetExitStatus as environment variable.